crates/miden-lib/asm/note_scripts/BURN.masm

```diff
@@ -0,0 +1,27 @@
+use.miden::contracts::faucets->faucet
+
+#! Network Faucet BURN script: burns the asset from the note by calling the network faucet's burn
+#! function. This note is intended to be executed against a network fungible faucet account.
```

Comment:
Both the basic and network fungible faucet have identical burn procedures, so is this specificity of the BURN note to the network faucet still accurate? Can I note send a BURN note to a basic fungible faucet, right? We are also calling `miden::contracts::faucets::burn` and not `miden::contracts::network_fungible::burn`. Their MAST roots are identical, but we basically call the interface (`faucets`) and not the specific network fungible contract, so to say (MASM doesn't have interfaces), but judging by the _name_, it should be possible to use the BURN note for any faucet that exposes `faucets::burn`.